src/logging.c

```diff
@@ -1501,3 +1514,52 @@ void rcutils_logging_console_output_handler(
     RCUTILS_SAFE_FWRITE_TO_STDERR("Failed to fini array.\n");
   }
 }
+
+static const char * expand_severity_with_color(
+  const logging_input_t * logging_input,
+  rcutils_char_array_t * logging_output,
+  size_t start_offset, size_t end_offset)
+{
+  (void)start_offset;
+  (void)end_offset;
+
+  if (rcutils_logging_is_colorized()) {
+    // The entire message is colorized, so we don't need to colorize the severity.
+    return expand_severity(logging_input, logging_output, start_offset, end_offset);
+  }
+
+  rcutils_ret_t status = RCUTILS_RET_OK;
+
+  SET_OUTPUT_COLOR_WITH_SEVERITY(status, logging_input->severity, *logging_output)
+  if (RCUTILS_RET_OK != status) {
+    RCUTILS_SAFE_FWRITE_TO_STDERR("Failed to set color on severity.\n");
+    return NULL;
+  }
+
+  const char * severity_string = g_rcutils_log_severity_names[logging_input->severity];
+  if (rcutils_char_array_strcat(logging_output, severity_string) != RCUTILS_RET_OK) {
+    RCUTILS_SAFE_FWRITE_TO_STDERR(rcutils_get_error_string().str);
+    rcutils_reset_error();
+    RCUTILS_SAFE_FWRITE_TO_STDERR("\n");
+    return NULL;
+  }
+
+  // If the severity is 4 characters long, add another space to line it up with the
+  // 5 character severities.
+  if (strlen(severity_string) == 4) {
+    if (rcutils_char_array_strcat(logging_output, " ") != RCUTILS_RET_OK) {
+      RCUTILS_SAFE_FWRITE_TO_STDERR(rcutils_get_error_string().str);
+      rcutils_reset_error();
+      RCUTILS_SAFE_FWRITE_TO_STDERR("\n");
+      return NULL;
+    }
+  }
```

Comment:
why we have this padding space only for this handler? i do not think this is consistent behavior for logging format?

Reply:
@ahcorde what do you think?